processing/src/main/java/org/apache/druid/query/groupby/GroupByQuery.java

```diff
@@ -121,6 +120,10 @@ public static Builder builder()
   @Nullable
   private final DateTime universalTimestamp;
 
+  private final boolean canPushDownLimit;
+  @Nullable
+  private Boolean forceLimitPushDown;
```

Comment:
Can we default to false instead of making this nullable?

```suggestion
  private boolean forceLimitPushDown;
```

Reply:
I think this would cause extra calls to `validateAndGetForceLimitPushDown` whenever `isApplyLimitPushdown` is called because of being unable to distinguish between an actual `false` and the value not being computed yet. It's probably not especially harmful, but it also seems unnecessary.

Reply:
Yeah, it's nullable to lazily initialize and cache it. Checking the value of `forceLimitPushDown` is needed a couple of times during a query. Even though it's not very harmful, I don't see any reason to not cache it.